src/Entity.ts

```diff
@@ -0,0 +1,25 @@
+import type { Observable } from 'rxjs'
+import type { Centrifuge } from './Centrifuge.js'
+import type { CentrifugeQueryOptions } from './types/query.js'
+
+export class Entity {
```

Comment:
curious why you opted to redefine functions on the class (_transact, _query, _transactSequence) instead of extending the Centrifuge class and just overriding where necessary e.g the query keys

Reply:
Extending the Centrifuge class for each entity would result in there being many instances of it. But there needs to be only one, because that's where the cache lives. the extra `_transact` and `_transactSequence` here are a small convenience so you can do, `this._transact()` instead of `this._root._transact()`. I don't think the functions themselves could be moved to Entity, because we'll need something like `Centrifuge.createPool()` at some point.